### DISCUSSION TITLE
How to apply Blur filter?

### DISCUSSION THREAD

**Discussion by H1manshu21** [Blaze-Stars/Image-Filter-Application#24] [Q&A] [answered]
Discuss algorithm on how to apply `blur` effect on `BMP` image.

**Comment by chandrakant100** [comment]
| Original Image and its Blured Image |
| ------------------------------------------- |
| <img src="https://coderwall-assets-0.s3.amazonaws.com/uploads/picture/file/497/blur.jpg" height="300px" width="700px"> |

**Comment by SHembram18** [comment]
By applying Blur filter to an image, The output of the image looks like little bit softer ( _Something that can't visible clearly_ )
There is an method called `Box Blur` .
By calculating the average value of the neighbour pixel (one row & one coloumn)

**Reply by Tushar7-coder** [reply]
Agree with @HembramBeta777

**Comment by Tushar7-coder** [comment, marked as answer]
## Pseudocode for Blur Filter
--> Make a temporary copy of the original picture to work with. 
--> For a certain pixel:-  (If it's not in the edge of the picture)-:
                                • Find the average of the 3x3 matrix with the pixel in question at the centre for R, G and B
                                • Store the three averages of R, G, B in the corresponding pixel of the temporary copy image.
                            :- If it's on the top edge (except corner pixels):
                               • Find the average of the 2x3 matrix with the pixel in question at the position (1,2) for R, G and B.
                               • Store the three averages of R, G, B in the corresponding pixel of the temporary copy image.
                            :- If it's on the bottom edge (except corner pixels):
                              • Find the average of the 2x3 matrix with the pixel in question at the position (2,2) for R, G and B.
                              • Store the three averages of R, G, B in the corresponding pixel of the temporary copy image. 
                            :- If it's on the left edge (except corner pixels):
                              • Find the average of the 3x2 matrix with the pixel in question at the position (2,1) for R, G and B. 
                              • Store the three averages of R, G, B in the corresponding pixel of the temporary copy image. 
                            :- If it's on the right edge (except corner pixels):
                             • Find the average of the 3x2 matrix with the pixel in question at the position (2,2) for R,Gand B.
                             • Store the three averages of R, G, B in the corresponding pixel of the temporary copy image. 
                            :- If it's on the top-left corner:
                             • Find the average of the 2x2 matrix with the pixel in question at the position (1,1) for R,G and B.
                             • Store the three averages of R, G, B in the corresponding pixel of the temporary copy image.

**Reply by chandrakant100** [reply]
Good Approach !!
